lib/sqewer/worker.rb

```diff
@@ -89,6 +89,10 @@ def start
     @logger.info { '[worker] Starting with %d consumer threads' % @num_threads }
     @execution_queue = Queue.new
 
+    # Ensure that unhandled exceptions inside threads make the worker fail,
+    # to avoid silent failures with no consumer threads running.
+    Thread.abort_on_exception = true
```

Comment:
I agree with the approach, but this could be potentially dangerous if used in different systems, so let's keep an eye open.

Reply:
Indeed, sqewer is used in other services like transfer-web, peek, and probably others I can't remember now

Reply:
Yep, I agree. At the same time, I think that the current setup –leading to zombie pods– could be dangerous in any system using this lib, so I think it's worth it to ensure such a case can't happen again

Reply:
Yes, I agree with you, what I meant is that those other services could rely on this behaviour as a "feature", not sure how, but it could be :).

Reply:
This is a breaking change. We'll need to update major version and update changelog. Not sure how this gem is being released though.

Btw this is a public, open-source Ruby gem.

Reply:
Improved tests and bumped the major version ✅